docs/framework/react/quick-start.md

```diff
@@ -3,54 +3,132 @@ id: quick-start
 title: Quick Start
 ---
 
-The bare minimum to get started with TanStack Form is to create a form and add a field. Keep in mind that this example does not include any validation or error handling... yet.
+TanStack Form is unlike most form libraries you've used before. It's designed for large-scale production usage, with a focus on type safety, performance, and developer experience.
+
+As a result, we've developed [a philosophy around the library's usage](/form/latest/docs/philosophy) that values scalability and long-term developer experience over short and sharable code snippets.
+
+Here's an example of a form following many of our best practices, which will allow you to rapidly develop even high-complexity forms after a short onboarding experience:
 
 ```tsx
 import React from 'react'
 import ReactDOM from 'react-dom/client'
-import { useForm } from '@tanstack/react-form'
+import { createFormHook, createFormHookContexts } from '@tanstack/react-form'
+import { TextField, NumberField, SubmitButton } from '~our-app/ui-library'
+// We also support Valibot, ArkType, and any other standard schema library
+import { z } from 'zod'
 
-export default function App() {
-  const form = useForm({
+const { fieldContext, useFieldContext, formContext, useFormContext } =
```

Comment:
`useFieldContext` and `useFormContext` are not used here - would they normally be used? If yes, does it fit the example? If no, maybe it might make sense to skip them here?